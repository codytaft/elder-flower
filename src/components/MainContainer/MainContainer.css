.nav-link-section {
  display: flex;
  flex-direction: column;
  text-align: center;
  margin: auto;
  font-size: 40px;
  font-family: 'Nunito', sans-serif;
  color: #453d30;
  height: 100%; }

.nav-link {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  padding: 2rem;
  border: 0.2rem solid #4e2d1d;
  color: #453d30;
  border-radius: 5rem;
  cursor: pointer;
  width: 20rem;
  background: #eacda3; }

.nav-link:hover {
  background-color: #c3fcf2; }

.nav-link-hidden {
  display: none; }
