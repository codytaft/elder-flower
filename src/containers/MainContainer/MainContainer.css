.nav-link-section {
  display: flex;
  flex-direction: column;
  text-align: center;
  margin: auto;
  font-size: 40px;
  font-family: 'Nunito', sans-serif;
  color: #4e2d1d; }

.nav-link {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  padding: 2rem;
  border: 0.2rem solid #4e2d1d;
  color: #4e2d1d;
  border-radius: 5rem;
  cursor: pointer;
  width: 20rem; }

.nav-link:hover {
  background-color: rgba(78, 45, 29, 0.1); }

.nav-link-hidden {
  display: none; }
