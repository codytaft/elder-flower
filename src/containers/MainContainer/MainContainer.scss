.nav-link-section {
  display: flex;
  flex-direction: column;
  text-align: center;
  margin: auto;
  font-size: 40px;
  font-family: 'Nunito', sans-serif;
  color: #453d30;
  height: 100%;
}

.nav-link {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  padding: 2rem;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #453d30;
  border-radius: 5rem;
  cursor: pointer;
  width: 20rem;
  background: #eacda3;
}

.nav-link:hover {
  background-color: #c3fcf2;
}

.nav-link-hidden {
  display: none;
}

// .nav-link:active {
//   display: none;
// }
