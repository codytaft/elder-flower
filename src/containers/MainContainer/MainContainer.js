import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './MainContainer.css';
export class MainContainer extends Component {
  render() {
    return (
      <section className="nav-link-section">
        <NavLink exact to="/login" className="nav-link nav-link-login">
          Login
        </NavLink>
        <NavLink exact to="/sign-up-home" className="nav-link nav-link-signup">
          SignUp
        </NavLink>
      </section>
    );
  }
}
