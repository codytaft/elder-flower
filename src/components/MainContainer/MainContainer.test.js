import React from 'react';
import { shallow } from 'enzyme';
import { MainContainer } from './MainContainer';

describe('MainContainer', () => {
  let mockLocation;
  let wrapper;
  beforeEach(() => {
    mockLocation = {
      pathname: '/',
      search: '',
      hash: ''
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if pathname does not equal /', () => {
    mockLocation = {
      pathname: '/home',
      search: '',
      hash: ''
    };
    wrapper = shallow(<MainContainer location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
