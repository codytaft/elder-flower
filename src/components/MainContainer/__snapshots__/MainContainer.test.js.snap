// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
        Object {
            "hash": "",
            "pathname": "/",
            "search": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link nav-link-login"
          exact={true}
          to="/login"
>
          LOGIN
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link nav-link-signup"
          exact={true}
          to="/sign-up-home"
>
          SIGNUP
</NavLink>,
      ],
      "className": "nav-link-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "LOGIN",
          "className": "nav-link nav-link-login",
          "exact": true,
          "to": "/login",
        },
        "ref": null,
        "rendered": "LOGIN",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "SIGNUP",
          "className": "nav-link nav-link-signup",
          "exact": true,
          "to": "/sign-up-home",
        },
        "ref": null,
        "rendered": "SIGNUP",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link nav-link-login"
            exact={true}
            to="/login"
>
            LOGIN
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link nav-link-signup"
            exact={true}
            to="/sign-up-home"
>
            SIGNUP
</NavLink>,
        ],
        "className": "nav-link-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "LOGIN",
            "className": "nav-link nav-link-login",
            "exact": true,
            "to": "/login",
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "SIGNUP",
            "className": "nav-link nav-link-signup",
            "exact": true,
            "to": "/sign-up-home",
          },
          "ref": null,
          "rendered": "SIGNUP",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`MainContainer should match the snapshot if pathname does not equal / 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    location={
        Object {
            "hash": "",
            "pathname": "/home",
            "search": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link-hidden"
          exact={true}
          to="/login"
>
          LOGIN
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-link-hidden"
          exact={true}
          to="/sign-up-home"
>
          SIGNUP
</NavLink>,
      ],
      "className": "nav-link-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "LOGIN",
          "className": "nav-link-hidden",
          "exact": true,
          "to": "/login",
        },
        "ref": null,
        "rendered": "LOGIN",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "SIGNUP",
          "className": "nav-link-hidden",
          "exact": true,
          "to": "/sign-up-home",
        },
        "ref": null,
        "rendered": "SIGNUP",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link-hidden"
            exact={true}
            to="/login"
>
            LOGIN
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-link-hidden"
            exact={true}
            to="/sign-up-home"
>
            SIGNUP
</NavLink>,
        ],
        "className": "nav-link-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "LOGIN",
            "className": "nav-link-hidden",
            "exact": true,
            "to": "/login",
          },
          "ref": null,
          "rendered": "LOGIN",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "SIGNUP",
            "className": "nav-link-hidden",
            "exact": true,
            "to": "/sign-up-home",
          },
          "ref": null,
          "rendered": "SIGNUP",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
